tsh.cc

```diff
@@ -36,16 +36,16 @@ static char prompt[] = "tsh> ";
 int verbose = 0;
 static char childprocs = 0;
 
-struct child 
+struct child
 {
-	short jobID = 0;
-	pid_t pid = 0;
-	string state = "";
-	string location = "";
-	string command = "";
+    short jobID = 0;
+    pid_t pid = 0;
+    char* state;
+    char* location;
+    char* command;
 } ;
 
-queue<child> jobHolder;
+deque<child*> jobHolder;
```

Comment:
Using a deque is better here because we get the best of both vectors and queues all in one.

Reply:
Also, we want to put these on heap instead of stack so lets use a pointer.